microservices/recordings/src/Service/Encoder.php

```diff
@@ -267,6 +271,20 @@ public function processAction(): void
                     ->setRecordedFileBaseName($baseName)
                     ->setRecordedFilePath($convertMp3);
 
+
+                $isDdi = $type === self::DDI && is_object($ddi);
+                $isUser = $type === self::ONDEMAND && is_object($user);
```

Comment:
$ddi and $user are actually objects or possible null values, what about?:

```suggestion
                $isDdi = $type === self::DDI && !is_null($ddi);
                $isUser = $type === self::ONDEMAND && !is_null($user);
```

Reply:
Those are "possible" undefined. I mean, if $type === self::DDI, then I'm sure $ddi is an object.
The same way., I'm sure that when $isUser === true $user is an object.

This check is only meant to provide some documentation for a pending (and scheduled) full refactor of this code. I rather keep the "is_object" because is more assertive but I could change it to is_null or removing it altogether, if you insist. Though I'm not sure which strategy is best in absence of tests and in case of a later modification: 
1) removing this, which could end with a fatal error.
2) keeping a check, which could result in a poorly functioning, but functional script (as in "it works, but stops storing the user or the ddi").